src/main/java/com/MyApp/budgetControl/domain/expense/ExpenseEntity.java

```diff
@@ -0,0 +1,57 @@
+package com.MyApp.budgetControl.domain.expense;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.EntityListeners;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import jakarta.validation.constraints.DecimalMin;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+
+import lombok.NoArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import java.time.Instant;
+import java.util.UUID;
+
+@Entity
+@Table(name = "expenses")
+@EntityListeners(AuditingEntityListener.class)
+@RequiredArgsConstructor
+@NoArgsConstructor(force = true)
+@Value
+public class ExpenseEntity {
+
+    public ExpenseEntity(ExpenseDTO expenseDTO) {
+        this.expenseId = UUID.randomUUID();
+        this.expenseCost = expenseDTO.getExpenseCost();
+        this.expenseCategory = expenseDTO.getExpenseCategory();
+        this.expenseComment = expenseDTO.getExpenseComment();
+        this.expenseDate = Instant.now();
+    }
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
```

Comment:
Right now You have 2 ways of setting id, through constructor by `this.expenseId = UUID.randomUUID()` and here through spring `@GeneratedValue(strategy = GenerationType.AUTO)` that's not correct, stick to one method where I would prefer id generated inside the app like You do in the constructor

Another technique is to have 2 different IDs - 1 technical generated through DB and 1 business generated by the app

Reply:
Updated the existing DTO to requestDTO for the POST method and added a new responseDTO.